api-server/src/server/boot/challenge.js

```diff
@@ -273,63 +272,55 @@ export function isValidChallengeCompletion(req, res, next) {
   return next();
 }
 
-export function modernChallengeCompleted(req, res, next) {
+export async function modernChallengeCompleted(req, res) {
```

Comment:
Express doesn't understand async handlers.  It doesn't `await` them, so error handling won't work properly.

We can either use a library to wrap async handlers or just write our own given how simple it is.

See https://expressjs.com/en/guide/error-handling.html for reference.

Reply:
Ah, thank you for that. Now it makes sense why rx used the `next` cb.

Reply:
No worries.

To be honest, I was a bit mystified by that, but I guess the second argument must be the error handler.

Reply:
Also, the same argument applies to https://github.com/freeCodeCamp/freeCodeCamp/pull/49386, but I didn't want to derail the discussion there.